dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -575,6 +588,7 @@ void DAGExpressionAnalyzer::buildAggGroupBy(
             /// need double check this assumption when we support agg with collation
             aggregation_keys.push_back(name);
             agg_key_set.emplace(name);
+            collators.push_back(nullptr);
```

Comment:
It seems L591 and L605 have changed the original logic. Was this intended?

Reply:
So, there was a bug in the previous code?

Reply:
The orignal implementation is no bug. I change this code to make sure the collators and `key_ref_agg_func` are corresponding one by one.